### ISSUE TITLE
web3.eth.handleRevert = true gives error -  reason: 'invalid arrayify value', code: 'INVALID_ARGUMENT', argument: 'value'

### ISSUE THREAD

**Issue by sagarparker** [ethereum/web3.js#4079]
I am trying to fetch require message from the smart contract : 

I am able to get the require error message on remix

I can also see the message on etherscan

**Web3 code :** 

const web3 = new Web3(rpcURL);

const Camps = require('../../../build/contracts/Camps.json');

const contract_address = process.env.camps_contract_address;

const abi = Camps.abi;

const contract = new web3.eth.Contract(abi,contract_address);

      router.post('/createCamp',
          body('camp_name').not().isEmpty(),
          body('camp_target').not().isEmpty(),
          body('camp_equity').not().isEmpty(),
          validateApiSecret,
          isAuthenticated,
          async(req,res)=>{
        try{

            web3.eth.handleRevert = true;

            //Input field validation
            const errors = validationResult(req);
            if (!errors.isEmpty()) {
                return res.status(422).json({
                    error: errors.array()[0],   
                });
            }

            const camp_name     =   req.body.camp_name;
            const camp_target   =   req.body.camp_target;
            const camp_equity   =   req.body.camp_equity;

            const txCount = await web3.eth.getTransactionCount(account_address);
            if(!txCount){
                return res.status(500).json({
                    result:false,
                    msg:'There was a problem creating a Camp'
                })
            }
            // Build the transaction
            const txObject = {
                nonce:    web3.utils.toHex(txCount),
                to:       contract_address,
                gasLimit: web3.utils.toHex(500000),
                gasPrice: web3.utils.toHex(web3.utils.toWei('1', 'gwei')),
                data: contract.methods.createCamp(camp_name,camp_target,camp_equity).encodeABI()
            }
        
            // Sign the transaction
            const tx = new Tx(txObject,{chain:42})
            tx.sign(privateKey)
        
            const serializedTx = tx.serialize()
            const raw = '0x' + serializedTx.toString('hex')

            // Broadcast the transaction
            await web3.eth.sendSignedTransaction(raw);

            return res.status(200).json({
                result:true,
                msg:'Camp created'
            });
            
        }
        catch(err){
            console.log(err);
            // console.log(web3.utils.hexToAscii(err.receipt.logsBloom));
            res.status(500).json({
                result:false,
                msg:'There was a problem creating a camp'
            })
        }
       });

**Error :** 

Error: invalid arrayify value (argument="value", value="0xx08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001343616d7020616c72656164792065786973747300000000000000000000000000", code=INVALID_ARGUMENT, version=bytes/5.1.0)    at Logger.makeError (D:\SagarWorkspace\Collective\backend\node_modules\@ethersproject\logger\lib\index.js:180:21)     
    at Logger.throwError (D:\SagarWorkspace\Collective\backend\node_modules\@ethersproject\logger\lib\index.js:189:20)    
    at Logger.throwArgumentError (D:\SagarWorkspace\Collective\backend\node_modules\@ethersproject\logger\lib\index.js:192:21)
    at Object.arrayify (D:\SagarWorkspace\Collective\backend\node_modules\@ethersproject\bytes\lib\index.js:91:19)        
    at AbiCoder.decode (D:\SagarWorkspace\Collective\backend\node_modules\@ethersproject\abi\lib\abi-coder.js:93:53)      
    at ABICoder.decodeParametersWith (D:\SagarWorkspace\Collective\backend\node_modules\web3-eth-abi\lib\index.js:304:30) 
    at ABICoder.decodeParameters (D:\SagarWorkspace\Collective\backend\node_modules\web3-eth-abi\lib\index.js:285:17)     
    at ABICoder.decodeParameter (D:\SagarWorkspace\Collective\backend\node_modules\web3-eth-abi\lib\index.js:274:17)      
    at sendTxCallback (D:\SagarWorkspace\Collective\backend\node_modules\web3-core-method\lib\index.js:513:50)
    at D:\SagarWorkspace\Collective\backend\node_modules\web3-core-requestmanager\lib\index.js:303:20
    at XMLHttpRequest.request.onreadystatechange (D:\SagarWorkspace\Collective\backend\node_modules\web3-providers-http\lib\index.js:98:13)
    at XMLHttpRequestEventTarget.dispatchEvent (D:\SagarWorkspace\Collective\backend\node_modules\xhr2-cookies\dist\xml-http-request-event-target.js:34:22)
    at XMLHttpRequest._setReadyState (D:\SagarWorkspace\Collective\backend\node_modules\xhr2-cookies\dist\xml-http-request.js:208:14)
    at XMLHttpRequest._onHttpResponseEnd (D:\SagarWorkspace\Collective\backend\node_modules\xhr2-cookies\dist\xml-http-request.js:318:14)
    at IncomingMessage.<anonymous> (D:\SagarWorkspace\Collective\backend\node_modules\xhr2-cookies\dist\xml-http-request.js:289:61)
    at IncomingMessage.emit (events.js:326:22) {
  reason: 'invalid arrayify value',
  code: 'INVALID_ARGUMENT',
  argument: 'value',
  value: '0xx08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001343616d7020616c72656164792065786973747300000000000000000000000000'
}

**Comment by luu-alex** [comment]
Hello, It seems like the value is incorrect as it should be hex, yours starts with '0xx', where is the value being assigned?

**Comment by sagarparker** [comment]
The v

> Hello, It seems like the value is incorrect as it should be hex, yours starts with '0xx', where is the value being assigned?

Hi Alex, the value is the msg being returned by the require function in the Solidity smart contract that I have written. Weird thing that I have observed is if I remove the 0x from the beginning and manually try to convert this to ASCII it returns something that was originally intended i.e "Camp not found". I tried checking the transaction on Etherscan it was returning the require error msg properly and also I tried to compile the smart contract on Remix there too it was returning the require error msg in the expected way. I am using Kovan test network and solidity ^0.8.0.
